python/kubeflow/trainer/backends/local_process.py

```diff
@@ -0,0 +1,209 @@
+from kubeflow.trainer.local import job
+
+
+logger = logging.getLogger(__name__)
+
+
+class LocalProcessBackend(base.TrainingBackend):
+
+    def __init__(self,
+                 cfg: LocalProcessBackendConfig,
+                 ):
+        # list of running subprocesses
+        self.__jobs: List[job.LocalJob] = []
+        self.cfg = cfg
+
+
+    def list_runtimes(self) -> List[types.Runtime]:
+        """List of the available Runtimes.
+
+               Returns:
+                   List[Runtime]: List of available local training runtimes.
+                       If no runtimes exist, an empty list is returned.
+               """
+
+        return local_runtimes.runtimes
+
+    def get_runtime(self, name: str) -> Optional[local_types.LocalRuntime]:
+        """Get the the Runtime object
+            Returns:
+                LocalRuntime: Runtime object for the given name.
+            Raises:
+                ValueError: If no Runtime is found for the given name.
+        """
+        _runtime = [rt for rt in local_runtimes.runtimes if rt.name == name]
+        if not _runtime:
+            raise ValueError(f"Runtime '{name}' not found.")
+
+        return _runtime[0]
+
+
+    def train(self, train_job_name:str,
+              runtime: local_types.LocalRuntime,
+              initializer: Optional[types.Initializer] = None,
+              trainer: Optional[types.Trainer] = None) -> str:
+        """
+                Create the LocalTrainJob. You can configure these types of training task:
+
+                - Custom Training Task: Training with a self-contained function that encapsulates
+                    the entire model training process, e.g. `CustomTrainer`.
+
+                Args:
+                    train_job_name: The name of the training job.
+                    runtime (`types.Runtime`): Reference to one of existing Runtimes.
+                    initializer (`Optional[types.Initializer]`):
+                        Configuration for the dataset and model initializers.
+                    trainer (`Optional[types.CustomTrainer]`):
+                        Configuration for Custom Training Task.
+
+                Returns:
+                    str: The unique name of the TrainJob that has been generated.
+
+                Raises:
+                    ValueError: Input arguments are invalid.
+                """
+        # Build the env
+        if not trainer:
+            raise ValueError("Cannot create TrainJob without a Trainer")
+        if isinstance(trainer, types.CustomTrainer):
+            trainer: types.CustomTrainer = trainer
+
+        # create temp dir for venv
+        target_dir = tempfile.mkdtemp(prefix=f"{train_job_name}-")
+        runtime.execution_dir = target_dir
+
+        # create venv
+        if runtime.create_venv:
+            self.__create_venv(env_dir=target_dir)
+            runtime.python_path = local_utils.get_venv_python_path(target_dir)
```

Comment:
Shall we always create a new venv for the Job submission ?
Why do we need `create_venv` flag ?

Reply:
@szaher Let's add this configuration later once we support Local runtime that doesn't require venv ?
For now, we only support single runtime: https://github.com/kubeflow/sdk/blob/71e83ae2897cd54aaca23ef5f1585e3577d31d3c/python/kubeflow/trainer/local/runtimes.py#L29